python_benchmarks/qr_codes/qr.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python
+
+"""qr.py
+Usage example:
+python qr.py -o out.yaml -p qrcodes/detection
+-H, --help - show help
+-o, --output - output file (default out.yaml)
+-p, --path - input dataset path (default qrcodes/detection)
+-a, --accuracy - input accuracy (default 47)
+-alg, --algorithm - input alg (default opencv)
+-m, --metric - input metric (default ~)
+"""
+
+import argparse
+import glob
+from enum import Enum
+
+import numpy as np
+import cv2 as cv
+
+
+class DetectorQR:
+    TypeDetector = Enum('TypeDetector', 'opencv opencv_wechat')
+    detected_corners = np.array([])
+    decoded_info = []
+    detector = None
+    type_detector = None
+
+    def __init__(self, type_detector=TypeDetector.opencv, path_to_model=""):
+        self.type_detector = type_detector
+        if type_detector == self.TypeDetector.opencv:
+            self.detector = cv.QRCodeDetector()
+        elif type_detector == self.TypeDetector.opencv_wechat:
+            self.detector = cv.wechat_qrcode_WeChatQRCode(path_to_model + "detect.prototxt",
+                                                          path_to_model + "detect.caffemodel",
+                                                          path_to_model + "sr.prototxt",
+                                                          path_to_model + "sr.caffemodel")
+        else:
+            raise TypeError("this type_detector isn't supported")
+
+    def detect(self, image):
+        if self.type_detector == self.TypeDetector.opencv:
```

Comment:
Why do you need single class with branches. Dedicated classes + duck typing solves it better.

Reply:
fixed